src/cudamatrix/cu-matrix.cc

```diff
@@ -2485,6 +2485,49 @@ void CuMatrixBase<Real>::Heaviside(const CuMatrixBase<Real> &src) {
   }
 }
 
+template<typename Real>
```

Comment:
BTW, if you are brave, you could consider making some kind of multi-line macro defined
inside cu-matrix.cc that will do this, so you could just replace this implementation with
```
// define CuMatrixBase<Real>::Exp(const CuMatrixBase<Real> &src) 
KALDI_DEFINE_CUDA_UNARY_FUNCTION(Exp, cuda_exp);  
```